src/libc/argdata/argdata_iterate.c

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2015 Nuxi, https://nuxi.nl/
+//
+// This file is distributed under a 2-clause BSD license.
+// See the LICENSE file for details.
+
+#include <common/argdata.h>
+
+#include <argdata.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+int argdata_map_iterate(const argdata_t *ad, argdata_iterator_t *it_) {
```

Comment:
Could you please place these functions in separate source files? Right now we're in the good spot that we're using a toolchain that automatically uses `-ffunction-sections`, `-fdata-sections` and `--gc-sections`, but I'd rather not depend on that too strongly. All other functions are currently placed in separate source files already.